### PR TITLE
Pages data view: Update quick-actions

### PR DESCRIPTION
**Pull request by jameskoster**
## What?
* Make 'Edit' a primary action so that it appears in the quick-action icon buttons.
* Swap the order so that 'Edit' is the first action, both in quick-actions and the ellipsis menu.

## Testing Instructions
1. Open the Pages data view.
2. Observe the Edit is is the first quick-action, and the first item in the ellipsis menu.

## Before
Screenshot 2024-03-14 at 09 52 23

## After

Screenshot 2024-03-14 at 15 00 21

### CODE REVIEW

**Comment by jasmussen** [comment]
Hmm, not sure about this one, actually. What this does is remove the quick-hover shortcut for trashing in favor of the edit button. But in the process, move to trash becomes the primary action in the ellipsis view, and the "Edit" action is already available when you just click the title of a post.

**Comment by jameskoster** [comment]
> the "Edit" action is already available when you just click the title of a post

Do you think this is clear enough? If a user were to assume that the designated Actions are the only ones available, it might seem strange that Edit isn't included there, and that trashing is given such prominence. 

On the latter point, trashing is not a super-common task for pages, so the ellipsis menu seems reasonable – like in Patterns.

> move to trash becomes the primary action in the ellipsis view

Good point, I'll flip trashing and viewing revisions.

cc @pablohoneyhoney since this change is based on some of your feedback.

**Comment by jasmussen** [comment]
> Good point, I'll flip trashing and viewing revisions.

That would help. I do personally like the ability to quick trash, but I also suspect this is something we can finetune.

File: packages/edit-site/src/components/actions/index.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ const { useHistory } = unlock( routerPrivateApis );
 export const trashPostAction = {
 	id: 'move-to-trash',
 	label: __( 'Move to Trash' ),
-	isPrimary: true,
+	isPrimary: false,
```
Comment (ntsekouras) [inline packages/edit-site/src/components/actions/index.js]: This is handled by default when not `true`, so we can remove this line completely. Additionally we can remove the `icon` too, as it's used only for primary actions.

**Review by ntsekouras — COMMENTED** [review]
Code wise this looks good and I left a really small comment. So it's just the design decision to be made 😄  Thanks!

**Comment by youknowriad** [comment]
Given that RC1 is today, I'm removing this from 6.5 as non essential.

**Comment by jameskoster** [comment]
Rebased and updated the OP to include before/after screenshots.

**Comment by oandregal** [comment]
I like the edit as an explicit action and I wouldn't mind trash being a secondary one. Two thoughts:

- having bulk editing available makes the difference between primary/secondary less valuable — e.g.: deleting a bunch of pages doesn't mean you have to open the menu for each of them, you can use bulk actions.
- unlike grid and list, the table layout doesn't have an stablished target for "edit", and, coupled with [this other change](https://github.com/WordPress/gutenberg/pull/59803) (reduces the title surface area) the title may be not clear enough.

If edit becomes primary for Pages, I'd expect to be so in all other pages as well (Templates, Parts, etc.). Essentially, for any page that uses the `table` layout.

The number of actions that are primary over the total seems too much. For example, 3 out of 4 actions are primary for posts that are not trashed, so it's easy to run into this:

Captura de ecrã 2024-03-14, às 13 05 57

All actions are primary for posts that are trashed:

Captura de ecrã 2024-03-14, às 13 12 42

**Comment by jameskoster** [comment]
> having bulk editing available makes the difference between primary/secondary less valuable — e.g.: deleting a bunch of pages doesn't mean you have to open the menu for each of them, you can use bulk actions.

Personally I agree that Trashing doesn't warrant being primary, for the reason you mention and simply because trashing is _relatively_ uncommon compared with other actions. 

> unlike grid and list, the table layout doesn't have an stablished target for "edit", and, coupled with https://github.com/WordPress/gutenberg/pull/59803 (reduces the title surface area) the title may be not clear enough.

Doesn't the explicit 'Edit' action help remedy that?

> If edit becomes primary for Pages, I'd expect to be so in all other pages as well (Templates, Parts, etc.).

I agree, and that's a detail to follow-up on if we want to merge this PR. Should that be something we add to the API? An `edit url` which other fields can reference, and causes the addition of the 'Edit' quick-action?

> The number of actions that are primary over the total seems too much. For example, 3 out of 4 actions are primary for posts that are not trashed, so it's easy to run into this:

I think we can massage things a bit in both cases. We already spoke about 'Move to Trash' potentially _not_ being a quick-action, and it's worth noting there will likely be other actions in the future e.g. 'Rename' and 'Duplicate'. Similarly in the Trash view 'Permanently delete' can be secondary as it's something more likely to be done in bulk.

I'll push some adjustments.

**Review by jasmussen — APPROVED** [review]
This seems like the right path forward, and it unlocks some followups including moving the destructive "revert revision" button into the ellipsis menu, as well as sorting similarly for other types like André suggests.

**Pull request merged**